src/test/java/net/openhft/chronicle/wire/WireResetTest.java

```diff
@@ -72,7 +70,7 @@ public void testDeepReset() {
         assertSame(identifier1, event1.identifier);
         assertNull(event1.identifier.id);
         assertTrue(event1.identifier.permissions.isEmpty());
-        assertNull(event1.identifier.parent.id);
+        assertNull(event1.identifier.parent);
```

Comment:
That's not a **deep** reset anymore. And is causing a GC.

Reply:
@yevgenp We need to return objects to their pristine (post-constructor) state. If post-constructed field value is null then we need to restore it to null also.
User can have different lifecycle for this field as to avoid GC.

Reply:
Note that it's not the only possibility, it's that we struggle to get team approval even from this toned down change. I encourage you to join the discussion on dev@.

Reply:
(note that in this case, resetting `parent` when child object is reset would definitely be a bug. Parent usually has multiple children and the subtree should not be all reset when only one child's `reset()` is called)